sourmash/commands.py

```diff
@@ -387,32 +391,35 @@ def index(args):
         if not ss:
             continue
 
-        # check to make sure we aren't loading incompatible signatures
-        if len(ksizes) > 1 or len(moltypes) > 1:
-            error('multiple k-mer sizes or molecule types present; fail.')
-            error('specify --dna/--protein and --ksize as necessary')
-            error('ksizes: {}; moltypes: {}',
-                  ", ".join(map(str, ksizes)), ", ".join(moltypes))
-            sys.exit(-1)
-
-        if nums == { 0 } and len(scaleds) == 1:
-            pass # good
-        elif scaleds == { 0 } and len(nums) == 1:
-            pass # also good
-        else:
-            error('trying to build an SBT with incompatible signatures.')
-            error('nums = {}; scaleds = {}', repr(nums), repr(scaleds))
-            sys.exit(-1)
-
+        check_signature_compatibilty_to_tree(ksizes, moltypes, nums, scaleds)
     notify('')
 
     # did we load any!?
     if n == 0:
         error('no signatures found to load into tree!? failing.')
         sys.exit(-1)
+    if return_n:
+        return tree, n
+    else:
+        return tree
 
-    notify('loaded {} sigs; saving SBT under "{}"', n, args.sbt_name)
-    tree.save(args.sbt_name, sparseness=args.sparseness)
+
+def check_signature_compatibilty_to_tree(ksizes, moltypes, nums, scaleds):
```

Comment:
in sourmash_args, we have `check_tree_is_compatible`; they do slightly different things and I think it's fine to use this function, but we should name it something else.

Reply:
(maybe prefix it with an _ to indicate that it's a helper function that shouldn't be used outside this module?)